Zend/zend_alloc.c

```diff
@@ -415,50 +415,77 @@ stderr_last_error(char *msg)
 /* OS Allocation */
 /*****************/
 
-#ifndef HAVE_MREMAP
-static void *zend_mm_mmap_fixed(void *addr, size_t size)
+static void zend_mm_munmap(void *addr, size_t size)
```

Comment:
Why were the locations of these method bodies swapped (e.g. munmap used to be below, and zend_mm_map_fixed was at this location on line 419)

This will make resolving merge conflicts when adding in patches harder, and make seeing the git history (e.g. `git blame`) harder.

If you need to call a function that's declared later, you can declare it above the definition (in the same ifdef checks, if any, for _WIN32 functions)

`static void *zend_mm_mmap_fixed(void *addr, size_t size);`

Reply:
`zend_mm_munmap` should be used in `zend_mm_mmap_fixed` (previously `munmap` and related error handling there was duplicated)

Reply:
Yes, I mean

```c
static void zend_mm_munmap(void *addr, size_t size); // declaration of method, e.g. in the top statements of the file

static void *zend_mm_mmap_fixed(void *addr, size_t size) { // definition of method in original location in the file, preserving git history
    /* ... call zend_mm_munmap */
}

static void zend_mm_munmap(void *addr, size_t size) { // definition of method in original location in the file, preserving git history
}
```

Reply:
the low-levels methods are modified a lot, basically rewritten, I prefer to not declare/prototype static method first, also, no allocation is expected to be done in free func later